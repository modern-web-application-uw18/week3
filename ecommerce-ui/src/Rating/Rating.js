import React, {Component} from 'react';
import PropTypes from 'prop-types';
import Host from '../Host/Host';
// import ReactRating from 'react-rating';

class Rating extends Component {

    render() {

        const {
            stars,
            reviews,
        } = this.props.rating;

        return ( <div> Rating: {this.props.rating.stars} &middot; Reviews: {this.props.rating.reviews} </div> );
    }
            
}

Rating.propTypes = {
    stars: PropTypes.number.isRequired,
    reviews: PropTypes.number.isRquired
}

Rating.defaultProps = {
    stars: 3,
    reviews: 0
}
export default Rating;