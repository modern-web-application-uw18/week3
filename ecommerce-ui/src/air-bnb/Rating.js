import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ReactRating from 'react-rating';
import greenstar from '../img/green-star.png';
import graystar from '../img/gray-star.png';

class Rating extends Component {

    render() {

        const {
            stars,
            reviews
        } = this.props.rating;

        return (
            <span>{reviews} reviews &middot;
                <ReactRating fractions={2} initialRating={stars}
                placeholderSymbol={<img src={greenstar} alt=""/>} fullSymbol={<img src={greenstar} alt=""/>}
                emptySymbol={<img src={graystar} alt=""/>}/></span>
        );
    }
}

Rating.propTypes = {
    stars: PropTypes.number.isRequired,
    reviews: PropTypes.number.isRequired
};

Rating.defaultProps = {
    stars: 3,
    reviews: 0
};

export default Rating;
