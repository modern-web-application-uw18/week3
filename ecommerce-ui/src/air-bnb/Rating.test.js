import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import Rating from './Rating';

describe('Rating', () => {
    it('renders without crashing', () => {
        const rating = {"stars": 3.5, "reviews": 675};
        const div = document.createElement('div');
        ReactDOM.render(<Rating rating={rating} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('should render with stars and reviews', () => {
        const rating = {"stars": 3, "reviews": 111};
        const component = TestRenderer.create(
            <Rating rating={rating}/>);
            expect(component.toJSON()).toMatchSnapshot();
    });
});