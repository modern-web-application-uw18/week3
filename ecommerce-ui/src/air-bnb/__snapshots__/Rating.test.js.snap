// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating should render with stars and reviews 1`] = `
<span>
  111
   reviews Â·
  <span
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "direction": "ltr",
        "display": "inline-block",
      }
    }
  >
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <img
        alt=""
        src="gray-star.png"
      />
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      >
        <img
          alt=""
          src="green-star.png"
        />
      </span>
    </span>
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <img
        alt=""
        src="gray-star.png"
      />
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      >
        <img
          alt=""
          src="green-star.png"
        />
      </span>
    </span>
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <img
        alt=""
        src="gray-star.png"
      />
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      >
        <img
          alt=""
          src="green-star.png"
        />
      </span>
    </span>
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <img
        alt=""
        src="gray-star.png"
      />
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "0%",
          }
        }
      >
        <img
          alt=""
          src="green-star.png"
        />
      </span>
    </span>
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <img
        alt=""
        src="gray-star.png"
      />
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "0%",
          }
        }
      >
        <img
          alt=""
          src="green-star.png"
        />
      </span>
    </span>
  </span>
</span>
`;
